File: frontend/src/features/issue/services/issueService.js
import api from "@/api/axios";

const BASE_URL = "/issues";

const issueService = {
    getIssues: async () => {
        const response = await api.get(BASE_URL);
        return response.data;
    },

    getIssueById: async (id) => {
        const response = await api.get(`${BASE_URL}/${id}`);
        return response.data;
    },

    createIssue: async (issue) => {
        const response = await api.post(BASE_URL, issue);
        return response.data;
    },

    updateIssue: async (id, issue) => {
        const response = await api.put(`${BASE_URL}/${id}`, issue);
        return response.data;
    },

    deleteIssue: async (id) => {
        const response = await api.delete(`${BASE_URL}/${id}`);
        return response.data;
    },
};

export default issueService;

File: backend/src/main/java/com/aayush/assetmanagement/issue/service/IssueService.java
package com.aayush.assetmanagement.issue.service;

import com.aayush.assetmanagement.issue.dto.IssueRequest;
import com.aayush.assetmanagement.issue.dto.IssueResponse;

import java.util.List;

public interface IssueService {

    IssueResponse createIssue(IssueRequest request);

    IssueResponse getIssueById(Long id);

    List<IssueResponse> getAllIssues();

    IssueResponse updateIssue(Long id, IssueRequest request);

    void deleteIssue(Long id);

}